tests/src/test/scala/com/nvidia/spark/rapids/ParquetWriterSuite.scala

```diff
@@ -160,6 +160,39 @@ class ParquetWriterSuite extends SparkQueryCompareTestSuite {
     }
   }
 
+  test("set maxRecordsPerFile with partition concurrently") {
+    val tempFile = File.createTempFile("maxRecords", ".parquet")
+
+    Seq(("40", 40), ("200", 80)).foreach{ case (maxRecordsPerFile, expectedRecordsPerFile) =>
+      val conf = new SparkConf()
+        .set("spark.sql.files.maxRecordsPerFile", maxRecordsPerFile)
+        .set("spark.sql.maxConcurrentOutputFileWriters", "30")
+      try {
+        SparkSessionHolder.withSparkSession(conf, spark => {
+          import spark.implicits._
+          val df = (1 to 1600).map(i => (i, i % 20)).toDF()
+          df
+            .repartition(1)
+            .write
+            .mode("overwrite")
+            .partitionBy("_2")
+            .parquet(tempFile.getAbsolutePath())
```

Comment:
this looks hacky:  first creating an empty file, then overwriting it with a directory containing parquet files and then relying on tempFile.delete not to care whether the original tempFile is a fie or a dir? Let us use Files.createTempDirectory https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html

Reply:
Not quite sure about it, I just follow the pattern of other test cases in this suite, such as: https://github.com/NVIDIA/spark-rapids/blob/branch-22.12/tests/src/test/scala/com/nvidia/spark/rapids/ParquetWriterSuite.scala#L40-L44

> Let us use Files.createTempDirectory https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html

We could file a follow-up ticket this if it is a more reasonable solution.

Reply:
> Will `tempFile.delete` delete all the contents recursively if it is a directory?

It won't according to https://docs.oracle.com/javase/8/docs/api/java/io/File.html#delete-- "If this pathname denotes a directory, then the directory must be empty in order to be deleted." you should see after running the test 


> I just follow the pattern of other test cases in this suite, such as: https://github.com/NVIDIA/spark-rapids/blob/branch-22.12/tests/src/test/scala/com/nvidia/spark/rapids/ParquetWriterSuite.scala#L40-L44

Let us institute a good pattern to follow :)

Reply:
I'll push an update.

Reply:
Hmm, another question is that I find there is no `delete` function for a temp directory. To delete it, I have to covert it to `File` firstly:
```scala
val tempFile = Files.createTempDirectory("maxRecords.parquet")
try {
    ...
} finally {
    tempFile.toFile.delete()
}
```
, which means that we are still using the https://docs.oracle.com/javase/8/docs/api/java/io/File.html#delete--

Reply:
given the current dependencies we could use:
-  org.apache.hadoop.fs.FileUtil.fullyDelete(tempFile) 
-  org.apache.commons.io.FileUtils.deleteDirectory(tempFile)
-  org.apache.spark.network.util.JavaUtils.deleteRecursively(tempFile)
- add a our own utility https://www.baeldung.com/java-delete-directory using Java8 API

Reply:
Update.